pkg/web/apps/apps_http_handler_test.go

```diff
@@ -88,3 +105,60 @@ func Test_HttpHandler_UpdateApp(t *testing.T) {
 		//assert.Equal(t, string(appJson)+"\n", rec.Body.String())
 	}
 }
+
+func Test_httpHandler_GetAppByID(t *testing.T) {
+	config := config.Get()
+  APIRoutePrefix := config.APIRoutePrefix
+	// make and configure a mocked Service
+	app := helpers.GetMockApp()
+	mockedAppService := &ServiceMock{
+		GetAppByIDFunc: func(ID string) (*models.App, error) {
+			if app != nil { 
+				return app, nil
+			}
+			return nil, models.ErrNotFound
+		},
+	}
+	// Setup
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.SetPath(APIRoutePrefix + "/apps/:id")
+	c.SetParamNames("id")
+	h := NewHTTPHandler(e, mockedAppService)
+	type fields struct {
+		Service Service
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		id      string
+		wantErr bool
+		want    *models.App
```

Comment:
The result of the handler is not the object. The want here should be a JSON/string

Reply:
covered in more detail below will close this comment